src/navigation.ts

```diff
@@ -554,6 +554,13 @@ export class Navigation {
     if (!cursor) {
       return;
     }
+
+    if (this.markedNode) {
+      cursor.setCurNode(this.markedNode);
+      this.removeMark(workspace);
+      return;
+    }
```

Comment:
From the description, this seems like a reasonable behavior change for this function. I think one nice to have would be to update the doc comment above to clarify the new special behavior for when a mark is present and how it relates to keepPosition's value (plus the general behavior of the function that's changed). 

Since we don't have tests, being clear in doc comments seems important to help increase the chance that behaviors are correctly copied over to the long-term implementation once we begin that work.

Reply:
Updated the comment--ptal